src/sets.jl

```diff
@@ -669,18 +669,53 @@ end
 
 Base.:(==)(set1::IndicatorSet{A, S}, set2::IndicatorSet{A, S}) where {A, S} = set1.set == set2.set
 
+"""
+    Complements(dimension::Int)
+
+The set corresponding to a mixed complementarity constraint.
+
+Complementarity constraints should be specified with an
+[`AbstractVectorFunction`](@ref)-in-`Complements(dimension)` constraint.
+
+The dimension of the vector-valued function `F` must be `2 * dimension`. This
+defines a complementarity constraint between the scalar function `F[i]` and the
+variable in `F[i + dimension]`. Thus, `F[i + dimension]` must be interpretable
+as a single variable `x_i` (e.g., `1.0 * x + 0.0`).
+
+If the variable `x_i` is constrained in `Interval(lb, ub)`, then mathematically,
+the mixed complementarity problem is to find a solution such that at least one
+of the following holds:
+
+  1.  F_i(x) = 0, lb <= x_i <= ub_i
+  2.  F_i(x) > 0, lb == x_i
+  3.  F_i(x) < 0,       x_i == ub_i
+
+Classically, the bounding set for `x_i` is `Interval(0, Inf)`, which recovers:
```

Comment:
What should the user do if the lower bound is not zero of the upper bound is not `Inf` ?

Reply:
```Julia
@variable(model, -1 <= x <= 1)
@constraint(model, [2 * x + 1, x] in Complements(1))
```
defines
```
2 * x + 1 = 0 or
2 * x + 1 > 0 if x == -1
2 * x + 1 < 0 if x == 1
```
which means the only solution is `2 * x + 1 = 0` so `x = -0.5`.

Reply:
Reading the docstring again, it seems clear and the reason I didn't understand it is due to a wrong preconception. Maybe adding the `SingleVariable`-in-`Interval` constraint in the Examples section would help.